img2dataset/downloader.py

```diff
@@ -75,9 +75,14 @@ def __init__(self, image_size, resize_mode, resize_only_if_bigger):
 
     def __call__(self, img_stream):
         try:
-            img = cv2.imdecode(np.frombuffer(img_stream.read(), np.uint8), 1)
+            img = cv2.imdecode(np.frombuffer(img_stream.read(), np.uint8), cv2.IMREAD_UNCHANGED)
             if img is None:
                 raise Exception("Image decoding error")
+            if len(img.shape) == 3 and img.shape[-1] == 4:
+                # replace transparent pixels to white - Source: https://stackoverflow.com/a/53737420/3474490
+                trans_mask = img[:,:,3] == 0
```

Comment:
This is going to leave some rough matting on certain PNGs. I don't thine cv2 color convert does alpha matting, does it? The issue with this is that it will work on certain PNGs, but some PNGS are not binary alpha masks and for this it will not work well. You should do a proper alpha matting with OpenCV to ensure that it handles semitransparent matting as well.

Reply:
Right now I'm just changing transparency from black to white. It's currently being converted to black.
I didn't understand which issue you were talking about, feel free to propose any improvement.
Do you have a sample image where this change is an issue?

Reply:
I tried on the image from https://github.com/np-csu/AlphaMatting

With or without this PR I'm getting this:
![000000001](https://user-images.githubusercontent.com/715491/142065349-f41a4dec-5fb5-46ad-9e1f-5898ad2c1c25.jpg)

Reply:
You wouldn't detect non-binary binary masks, you would just do proper alpha matting by multiplying the RGB by the alpha channel and multiplying the bg color by (1 - alpha) and summing them together. (assumes normalized 0-1 alpha).

Reply:
Thanks, I applied your suggestions